lib/services/resource.go

```diff
@@ -972,6 +977,26 @@ func UnmarshalProtoResource[T ProtoResourcePtr[U], U any](data []byte, opts ...M
 	return resource, nil
 }
 
+// UnmarshalProtoResourceArray unmarshals an array of ProtoResources from JSON using [UnmarshalProtoResource] on each
+// individual element.
+func UnmarshalProtoResourceArray[T ProtoResourcePtr[U], U any](data []byte, opts ...MarshalOption) ([]T, error) {
+	var msgs []json.RawMessage
+	if err := json.Unmarshal([]byte(data), &msgs); err != nil {
```

Comment:
This will make a copy of `data`, do we care?

Reply:
I don't believe so.  Outside of non-perf-critical CLI stuff, I don't know of a scenario where we might need to decode a non-proto sequence of proto resources.